dbt/adapters/duckdb/environments/local.py

```diff
@@ -100,20 +106,27 @@ def load_source(self, plugin_name: str, source_config: utils.SourceConfig):
                 params.append(source_config.database)
             if cursor.execute(q, params).fetchone()[0]:
                 if save_mode == "error_if_exists":
-                    raise Exception(f"Source {source_config.table_name()} already exists!")
+                    raise Exception(
+                        f"Source {source_config.table_name()} already exists!"
+                    )
                 else:
                     # Nothing to do (we ignore the existing table)
                     return
         df = plugin.load(source_config)
         assert df is not None
-        materialization = source_config.meta.get("materialization", "table")
-        cursor.execute(
-            f"CREATE OR REPLACE {materialization} {source_config.table_name()} AS SELECT * FROM df"
-        )
+
+        if plugin_name not in ["delta"]: # plugins which configure cursor itselfs
```

Comment:
I'm still not thrilled about special-casing the `delta` plugin like this; would it be better if we simply maintained the dictionary of source-created data frames (or dataframe-like equivalents) in the `Environment` instance, instead of inside of the `Plugin` class, for all of the sources? Would that work?

Reply:
In my opinion, we have to distinguish between materialization. Correct me if I am wrong
If you create table as df then you have the df instance and you copy data into duckdb process, if you create a view as df then you hold just the representation and pointer to df and load the first time when you resolve this view. 
This is why other plugins work with table materialization and without each cursor configuration. 

Therefore I am not sure if the idea is good to hold every instance of df in a dictionary in the Environment. Maybe I could make it so that we return a dataframe from the load function and if the materialization is a view register it to Environment. This could work and would solve also other plugins problems with the materialization view. 

Let me try it! Just thought about it while answerinng 😁

Reply:
Hi @jwills i did refactor now, can you take another look?
So i moved the logic into environment and abstracted it over the plugins if the plugin sets materialization to view
What i would like to do but i am not sure where exactly is to set default materialization for delta framework to view because it is per default better but it is not so much important